core/models.py

```diff
@@ -82,6 +83,26 @@ def display_with_agent_unit(self):
         )
 
 
+class FinSuiviContact(models.Model):
```

Comment:
Je ne suis pas sur que cette modélisation soit la bonne approche, pourquoi ne pas exploiter le many to many entre la fiche et les contacts pour utiliser la "through table" ?

Reply:
Au début je suis parti sur l'utilisation du many to many entre la fiche détection et les contacts et j'ai utilisé la "through table".
Mais je pense que cette solution n'est pas assez générique. Elle est valable pour FicheDetection uniquement et pas les autres types de fiche.
Je pense que l'état `fin de suivi` doit pouvoir s'appliquer sur n'importe quel type de fiche.
C'est ce qui m'a amener à modéliser cette classe `FinSuiviContact` et a utiliser `content_type`.

Reply:
Est ce qu'on est sur que le concept de fin de suivi va aussi s'appliquer dans les autre domaine ?
Je ne vois pas pourquoi cette solution n'est pas assez générique :thinking: , on s'en parle de vive voix dans la journée ?

Reply:
> Est ce qu'on est sur que le concept de fin de suivi va aussi s'appliquer dans les autre domaine ?

Je vais voir avec le métier mais je pense que oui.

> Je ne vois pas pourquoi cette solution n'est pas assez générique

Sauf erreur de ma part, il faut une classe comme `FicheDetectionContact` pour chaque fiche non ?

```
class FicheDetection(models.Model):
    # Autres champs...
    contacts = models.ManyToManyField(Contact, through='FicheDetectionContact', verbose_name="Contacts", blank=True)

class FicheDetectionContact(models.Model):
    fiche_detection = models.ForeignKey(FicheDetection, on_delete=models.CASCADE)
    contact = models.ForeignKey(Contact, on_delete=models.CASCADE)
    fin_suivi = models.DateField(null=True, blank=True) 

    class Meta:
        unique_together = ('fiche_detection', 'contact')
```

> on s'en parle de vive voix dans la journée ?

Oui :)